be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -168,84 +180,111 @@ object DbActor extends AskPatternConstants {
       channelDb
     }
 
-    override def receive: Receive = LoggingReceive {
-      case Write(channel, message) =>
-        log.info(s"Actor $self (db) received a WRITE request on channel '$channel'")
+    private def write(channel: Channel, message: Message): DbActorMessage = {
+      val channelDb: DB = channelsMap.get(channel) match {
+        case Some(channelDb) => channelDb
+        case _ => createDatabase(channel)
+      }
 
-        val channelDb: DB = channelsMap.get(channel) match {
-          case Some(channelDb) => channelDb
-          case _ => createDatabase(channel)
-        }
+      val messageId: Hash = message.message_id
+      Try(channelDb.put(messageId.getBytes, message.toJsonString.getBytes)) match {
+        case Success(_) =>
+          log.debug(s"Actor $self (db) wrote message_id '$messageId' on channel '$channel'")
+          DbActorWriteAck
+        case Failure(exception) =>
+          log.debug(s"Actor $self (db) encountered a problem while writing message_id '$messageId' on channel '$channel'")
+          DbActorNAck(ErrorCodes.SERVER_ERROR.id, exception.getMessage)
+      }
+    }
 
-        val messageId: Hash = message.message_id
-        Try(channelDb.put(messageId.getBytes, message.toJsonString.getBytes)) match {
-          case Success(_) =>
-            log.info(s"Actor $self (db) wrote message_id '$messageId' on channel '$channel'")
-            sender ! DbActorWriteAck
-          case Failure(exception) =>
-            log.info(s"Actor $self (db) encountered a problem while writing message_id '$messageId' on channel '$channel'")
-            sender ! DbActorNAck(ErrorCodes.SERVER_ERROR.id, exception.getMessage)
+    private def read(channel: Channel, messageId: Hash): DbActorMessage = {
+      if (channelsMap.contains(channel)) {
+        val channelDb: DB = channelsMap(channel)
+        Try(channelDb.get(messageId.getBytes)) match {
+          case Success(bytes) if bytes != null =>
+            DbActorReadAck(Some(Message.buildFromJson(new String(bytes, StandardCharsets.UTF_8))))
+          case _ =>
+            DbActorNAck(ErrorCodes.SERVER_ERROR.id, "Unknown read database error")
         }
+      } else {
+        DbActorReadAck(None)
+      }
```

Comment:
This might be easier to read: 
```suggestion
      if (!channelsMap.contains(channel)) {
        DbActorReadAck(None)
      } else {
        val channelDb: DB = channelsMap(channel)
        Try(channelDb.get(messageId.getBytes)) match {
          case Success(bytes) if bytes != null =>
            DbActorReadAck(Some(Message.buildFromJson(new String(bytes, StandardCharsets.UTF_8))))
          case _ =>
            DbActorNAck(ErrorCodes.SERVER_ERROR.id, "Unknown read database error")
        }
      }
```

also, I'm thinking that this line 
`DbActorReadAck(Some(Message.buildFromJson(new String(bytes, StandardCharsets.UTF_8))))`
could benefit from being split into multiple lines for improved readability :)